core/engine/src/value/macros.rs

```diff
@@ -0,0 +1,61 @@
+//! Macros related to `JsValue`.
+
+/// Create a `JsValue` from a simple DSL that resembles JSON.
+///
+/// ```
+/// # use boa_engine::{js_value, Context, JsValue};
+/// # let context = &mut Context::default();
+/// assert_eq!(js_value!( 1 ), JsValue::from(1));
+/// assert_eq!(js_value!( false ), JsValue::from(false));
+/// // Objects and arrays cannot be compared with simple equality.
+/// // To create arrays and objects, the context need to be passed in.
+/// assert!(js_value!([ 1, 2, 3 ], context).to_string(context), "[1,2,3]");
+///
+/// js_value!({
+///   //
+/// }, context);
+/// ```
+#[macro_export]
+macro_rules! js_value {
+    // Single pattern to simplify documentation.
+    ($($v:tt)+) => {
+        $crate::js_value_internal!($($v)+)
+    };
+}
```

Comment:
Could we also have an equivalent `js_object` macro that skips wrapping the object with `JsValue`? Seems useful to have this alternative version for when someone just needs the object itself instead of a value.

Reply:
Another idea I had was to make `js_value!` private and only expose `js_object`, since `JsValue::from(f32::INFINITY)` is only 5 characters longer than using the macro. This would completely sidestep tje problem of having to surround somve special values like `f32::INFINITY` with parentheses.